From: QbProg <tholag@gmail.com>
Date: Thu, 12 May 2011 21:03:41 +0200
Subject: [PATCH 59/82] [warning-fix][unreachable-code]

Fixes unreachable code warnings
---
 src/FSD/FSD_CmpFile.cxx |    2 +-
 src/FSD/FSD_File.cxx    |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sat, 11 Jun 2011 20:04:54 +0200
Subject: [PATCH 60/82] [warning-fix][unreachable-code]

Fixed some unreachable code warnings
---
 .../APIHeaderSection_EditHeader.cxx                |    2 +-
 src/DDataStd/DDataStd_DrawDriver.cxx               |   10 ----------
 src/DPrsStd/DPrsStd_AISPresentationCommands.cxx    |    2 --
 src/Draw/Draw_Window.cxx                           |    2 ++
 .../GeometryTest_ConstraintCommands.cxx            |    2 ++
 src/XSControl/XSControl_Functions.cxx              |    3 +++
 6 files changed, 8 insertions(+), 13 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sun, 12 Jun 2011 12:09:03 +0200
Subject: [PATCH 61/82] [warning-fix][unreachable-code]

Removed unreachable code warnings, and marked with @todo when in doubt
---
 src/BRepFill/BRepFill_OffsetWire.cxx               |    4 +++-
 src/BRepFill/BRepFill_Pipe.cxx                     |    2 +-
 src/ChFi2d/ChFi2d_Builder_0.cxx                    |    2 +-
 src/ChFi3d/ChFi3d_Builder_0.cxx                    |    2 +-
 src/ChFi3d/ChFi3d_Builder_CnCrn.cxx                |    2 +-
 src/ChFi3d/ChFi3d_Builder_SpKP.cxx                 |    4 ++--
 src/Geom2dGcc/Geom2dGcc_Circ2d2TanOn.cxx           |    4 ++++
 src/Geom2dGcc/Geom2dGcc_Circ2d2TanRad.cxx          |    4 ++++
 src/GeomFill/GeomFill.cxx                          |    5 +++++
 src/GeomFill/GeomFill_LocationGuide.cxx            |    4 +++-
 src/IntPatch/IntPatch_LineConstructor.cxx          |    4 ++++
 src/IntPatch/IntPatch_Polyhedron.cxx               |    5 ++++-
 src/IntSurf/IntSurf_Quadric.cxx                    |    8 ++++++++
 src/TopOpeBRep/TopOpeBRep_ShapeIntersector2d.cxx   |    2 +-
 src/TopOpeBRep/TopOpeBRep_sort.cxx                 |    2 +-
 src/TopOpeBRepBuild/TopOpeBRepBuild_Builder1.cxx   |    3 +++
 src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx      |    6 ++++--
 src/TopOpeBRepBuild/TopOpeBRepBuild_Tools_1.cxx    |    5 +----
 src/TopOpeBRepDS/TopOpeBRepDS.cxx                  |    2 +-
 src/TopOpeBRepDS/TopOpeBRepDS_PointIterator.cxx    |    6 ++++++
 .../TopOpeBRepTool_makeTransition.cxx              |    3 ++-
 src/TopOpeBRepTool/TopOpeBRepTool_mkTondgE.cxx     |    5 ++++-
 22 files changed, 64 insertions(+), 20 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Fri, 17 Jun 2011 21:08:22 +0200
Subject: [PATCH 62/82] [warning-fix][unreachable-code]

Fixed unreachable code warnings
---
 inc/GCPnts_UniformDeflection.gxx                   |    2 ++
 .../ProjLib_ComputeApproxOnPolarSurface.cxx        |    2 ++
 2 files changed, 4 insertions(+), 0 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sun, 19 Jun 2011 18:01:54 +0200
Subject: [PATCH 63/82] [warning-fix][unreachable-code] Fixes unreachable code
 warnings

---
 inc/GCPnts_QuasiUniformDeflection.gxx              |    3 +++
 src/AIS/AIS_Selection.cxx                          |    6 ++++--
 src/Adaptor3d/Adaptor3d_IsoCurve.cxx               |    2 +-
 .../Adaptor3d_SurfaceOfLinearExtrusion.cxx         |    2 +-
 src/Adaptor3d/Adaptor3d_SurfaceOfRevolution.cxx    |    2 +-
 src/BOP/BOP_CorrectTolerances.cxx                  |    3 ---
 src/BOPTools/BOPTools_PaveFiller_1.cxx             |    5 ++++-
 src/BRepGProp/BRepGProp_EdgeTool.cxx               |    2 +-
 .../BRepTools_NurbsConvertModification.cxx         |    2 ++
 src/Graphic2d/Graphic2d_GraphicObject.cxx          |    3 ++-
 src/Graphic2d/Graphic2d_SetOfMarkers.cxx           |    3 ++-
 src/IntTools/IntTools_EdgeFace.cxx                 |    2 ++
 src/MAT2d/MAT2d_Circuit.cxx                        |    2 ++
 src/MNaming/MNaming_NamingRetrievalDriver.cxx      |    2 +-
 src/MNaming/MNaming_NamingRetrievalDriver_1.cxx    |    2 +-
 src/OpenGl/OpenGl_GraphicDriver_9.cxx              |   16 +++++++++-------
 src/OpenGl/OpenGl_togl_print.cxx                   |    3 ++-
 src/ShapeAnalysis/ShapeAnalysis.cxx                |    2 ++
 src/ShapeBuild/ShapeBuild_Edge.cxx                 |    2 ++
 .../ShapeConstruct_ProjectCurveOnSurface.cxx       |    2 ++
 src/ShapeCustom/ShapeCustom_BSplineRestriction.cxx |    2 ++
 src/ShapeFix/ShapeFix_EdgeProjAux.cxx              |    2 ++
 src/ShapeFix/ShapeFix_Wireframe.cxx                |    2 --
 src/StdPrs/StdPrs_Curve.cxx                        |    6 +++++-
 src/StdPrs/StdPrs_DeflectionCurve.cxx              |    2 ++
 src/StdSelect/StdSelect_BRepSelectionTool.cxx      |    5 +++--
 26 files changed, 58 insertions(+), 27 deletions(-)

From a31a091d79aa35deb85b4046b67ca8e4df495165 Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sat, 19 May 2012 16:14:56 +0200
Subject: [PATCH] [warning-fix][unreachable-code] Fixes unreachable code
 warnings

---
 src/TopOpeBRep/TopOpeBRep_FFDumper.cxx      |    5 +++--
 src/TopOpeBRep/TopOpeBRep_traceBOOPNINT.cxx |    6 ++----
 src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx |    8 ++------
 3 files changed, 7 insertions(+), 12 deletions(-)

Index: oce/src/GCPnts/GCPnts_QuasiUniformDeflection.gxx
===================================================================
--- oce.orig/src/GCPnts/GCPnts_QuasiUniformDeflection.gxx
+++ oce/src/GCPnts/GCPnts_QuasiUniformDeflection.gxx
@@ -229,7 +229,10 @@ static Standard_Boolean PerformComposite
 
     Ua = Ub;
   }
+ 
+#ifndef _MSC_VER
   return Standard_True;
+#endif
 }
 
 
Index: oce/src/GCPnts/GCPnts_UniformDeflection.gxx
===================================================================
--- oce.orig/src/GCPnts/GCPnts_UniformDeflection.gxx
+++ oce/src/GCPnts/GCPnts_UniformDeflection.gxx
@@ -194,7 +194,9 @@ static Standard_Boolean PerformComposite
 
     Ua = Ub;
   }
+  #ifndef _MSC_VER
   return Standard_True;
+  #endif
 }
 
 
Index: oce/src/AIS/AIS_Selection.cxx
===================================================================
--- oce.orig/src/AIS/AIS_Selection.cxx
+++ oce/src/AIS/AIS_Selection.cxx
@@ -285,8 +285,10 @@ AIS_SelectStatus AIS_Selection::Select(c
   S->myResultMap.Bind( anObject, aListIter );
   return AIS_SS_Added;
 #endif //USE_MAP
-  return AIS_SS_Removed;
-  
+
+#ifndef _MSC_VER
+  return AIS_SS_Removed;  
+#endif
 }
 
 //=======================================================================
Index: oce/src/APIHeaderSection/APIHeaderSection_EditHeader.cxx
===================================================================
--- oce.orig/src/APIHeaderSection/APIHeaderSection_EditHeader.cxx
+++ oce/src/APIHeaderSection/APIHeaderSection_EditHeader.cxx
@@ -38,7 +38,7 @@ static Standard_Boolean IsTimeStamp
     case  4 : if (uncar < '0' || uncar > '9') return Standard_False; break;
     case  5 : if (uncar != '-') return Standard_False;
     case  6 : if (uncar != '0' && uncar != '1') return Standard_False;  break;
-    case  7 : if (uncar < '0' || uncar > '9') return Standard_False; break;
+    case  7 : if (uncar < '0' || uncar > '9') return Standard_False; 
       if (dizmois == '1' && (uncar < '0' || uncar > '2')) return Standard_False;
       break;
     case  8 : if (uncar != '-') return Standard_False;
Index: oce/src/Adaptor3d/Adaptor3d_IsoCurve.cxx
===================================================================
--- oce.orig/src/Adaptor3d/Adaptor3d_IsoCurve.cxx
+++ oce/src/Adaptor3d/Adaptor3d_IsoCurve.cxx
@@ -805,7 +805,7 @@ gp_Elips Adaptor3d_IsoCurve::Ellipse() c
   default:  
     {
       Standard_NoSuchObject::Raise("Adaptor3d_IsoCurve:Ellipse");
-      return gp_Elips();
+      break;
     }
     
   }
Index: oce/src/Adaptor3d/Adaptor3d_SurfaceOfLinearExtrusion.cxx
===================================================================
--- oce.orig/src/Adaptor3d/Adaptor3d_SurfaceOfLinearExtrusion.cxx
+++ oce/src/Adaptor3d/Adaptor3d_SurfaceOfLinearExtrusion.cxx
@@ -489,7 +489,7 @@ GeomAbs_SurfaceType Adaptor3d_SurfaceOfL
     }
 
   default:
-    return GeomAbs_SurfaceOfExtrusion;
+    break;
 
   }
   
Index: oce/src/BRepFill/BRepFill_Pipe.cxx
===================================================================
--- oce.orig/src/BRepFill/BRepFill_Pipe.cxx
+++ oce/src/BRepFill/BRepFill_Pipe.cxx
@@ -633,7 +633,7 @@ TopoDS_Shape BRepFill_Pipe::MakeShape(co
     return solid;
   }
   else {
-    return result;
+    
   }
   return result;
 }
Index: oce/src/BRepGProp/BRepGProp_EdgeTool.cxx
===================================================================
--- oce.orig/src/BRepGProp/BRepGProp_EdgeTool.cxx
+++ oce/src/BRepGProp/BRepGProp_EdgeTool.cxx
@@ -63,7 +63,7 @@ Standard_Integer  BRepGProp_EdgeTool::In
     break;
     
     default :
-      return 10;
+      break;
   }
   //POP WE should return something : -> Default
   return 10;
Index: oce/src/BRepTools/BRepTools_NurbsConvertModification.cxx
===================================================================
--- oce.orig/src/BRepTools/BRepTools_NurbsConvertModification.cxx
+++ oce/src/BRepTools/BRepTools_NurbsConvertModification.cxx
@@ -598,8 +598,10 @@ Standard_Boolean BRepTools_NurbsConvertM
       return Standard_True;
     }
   }
+#ifndef _MSC_VER
   Curve2d = C2d;
   return Standard_True;
+#endif
 }
 
 //=======================================================================
Index: oce/src/ChFi2d/ChFi2d_Builder_0.cxx
===================================================================
--- oce.orig/src/ChFi2d/ChFi2d_Builder_0.cxx
+++ oce/src/ChFi2d/ChFi2d_Builder_0.cxx
@@ -705,7 +705,7 @@ gp_Pnt ComputePoint(const TopoDS_Vertex&
       Param = computePoint.Parameter();
     }
     thePoint = cc.Value(Param);
-    return thePoint;
+    
   } // else ...
   return thePoint;
 } // ComputePoint
Index: oce/src/ChFi3d/ChFi3d_Builder_CnCrn.cxx
===================================================================
--- oce.orig/src/ChFi3d/ChFi3d_Builder_CnCrn.cxx
+++ oce/src/ChFi3d/ChFi3d_Builder_CnCrn.cxx
@@ -915,7 +915,7 @@ static Standard_Integer SurfIndex(const
   case ChFiSURFACE: return aSurfData->Surf();
   case FACE1:       return aSurfData->IndexOfS1();
   case FACE2:       return aSurfData->IndexOfS2();
-  default:          return -1;
+  default:          break;
   }
   return -1;
 }
Index: oce/src/ChFi3d/ChFi3d_Builder_SpKP.cxx
===================================================================
--- oce.orig/src/ChFi3d/ChFi3d_Builder_SpKP.cxx
+++ oce/src/ChFi3d/ChFi3d_Builder_SpKP.cxx
@@ -197,7 +197,7 @@ static Standard_Real ComputeAbscissa(con
   case GeomAbs_Circle:
     return C.Circle().Radius()*U;
   default:
-    return 0;
+    break;
   }    
   Standard_NotImplemented::Raise("calculate abscisse non-processed");
   return 0.;
@@ -450,7 +450,7 @@ static Standard_Real ResetProl(const Top
   const Handle(Geom_Surface)& surf = DStr.Surface(CD->Surf()).Surface();
   Standard_Real par = 0., x, y;
   if(!isfirst) par = edglen;
-  Standard_Real sppar;
+  Standard_Real sppar = 0.0;
   for (Standard_Integer i = 1; i <= 2; i++) { 
     CD->ChangeInterference(i).SetParameter(par,isfirst);
     Handle(Geom2d_Curve) pc = CD->Interference(i).PCurveOnSurf();
Index: oce/src/DDataStd/DDataStd_DrawDriver.cxx
===================================================================
--- oce.orig/src/DDataStd/DDataStd_DrawDriver.cxx
+++ oce/src/DDataStd/DDataStd_DrawDriver.cxx
@@ -249,16 +249,6 @@ Handle(Draw_Drawable3D) DDataStd_DrawDri
       }
       else {
 	break;
-	TopoDS_Shape aLocalShape = (Geometry (A, 1, TopAbs_FACE));
-	TopoDS_Face F1 = TopoDS::Face (aLocalShape);
-//	TopoDS_Face F1 = TopoDS::Face (Geometry (A, 1, TopAbs_FACE));
-	if (A->NbGeometries() == 1) D = new DrawDim_Distance (F1);
-	else {
-	  TopoDS_Shape aLocalShape = Geometry(A,2,TopAbs_FACE);
-	  TopoDS_Face F2 = TopoDS::Face(aLocalShape);
-//	  TopoDS_Face F2 = TopoDS::Face(Geometry(A,2,TopAbs_FACE));
-	  D = new DrawDim_Distance(F1,F2);
-	}
       }
     }
     break;
Index: oce/src/DPrsStd/DPrsStd_AISPresentationCommands.cxx
===================================================================
--- oce.orig/src/DPrsStd/DPrsStd_AISPresentationCommands.cxx
+++ oce/src/DPrsStd/DPrsStd_AISPresentationCommands.cxx
@@ -89,8 +89,6 @@ static Standard_Integer DPrsStd_AISDispl
     TPrsStd_AISViewer::Update(L);
     return 0;
   }
-  di << "DPrsStd_AISDisplay : Error" << "\n";
-  return 1;
 }
 
 //=======================================================================
Index: oce/src/Draw/Draw_Window.cxx
===================================================================
--- oce.orig/src/Draw/Draw_Window.cxx
+++ oce/src/Draw/Draw_Window.cxx
@@ -1398,7 +1398,9 @@ LONG APIENTRY DrawWindow::DrawProc(HWND
     else
       return LONG(DefMDIChildProc(hWnd, wMsg, wParam, lParam));
   }
+#ifndef _MSC_VER // unreachable code in MSVC
   return (0l);
+#endif
 }
 
 
Index: oce/src/FSD/FSD_CmpFile.cxx
===================================================================
--- oce.orig/src/FSD/FSD_CmpFile.cxx
+++ oce/src/FSD/FSD_CmpFile.cxx
@@ -1392,7 +1392,7 @@ Storage_Position FSD_CmpFile::Tell()
     else
       return aPosR;
   }
-  default: return -1;
+  default: break;
   }
   return -1;
 }
Index: oce/src/FSD/FSD_File.cxx
===================================================================
--- oce.orig/src/FSD/FSD_File.cxx
+++ oce/src/FSD/FSD_File.cxx
@@ -1366,7 +1366,7 @@ Storage_Position FSD_File::Tell()
     else
       return aPosR;
   }
-  default: return -1;
+  default: break;
   }
   return -1;
 }
Index: oce/src/Geom2dGcc/Geom2dGcc_Circ2d2TanOn.cxx
===================================================================
--- oce.orig/src/Geom2dGcc/Geom2dGcc_Circ2d2TanOn.cxx
+++ oce/src/Geom2dGcc/Geom2dGcc_Circ2d2TanOn.cxx
@@ -715,7 +715,9 @@ Standard_Boolean Geom2dGcc_Circ2d2TanOn:
     if (TheSame1(Index) == 0) { return Standard_False; }
     else { return Standard_True; }
   }
+#ifndef _MSC_VER
   return Standard_False;
+#endif 
 }
 
 Standard_Boolean Geom2dGcc_Circ2d2TanOn::
@@ -731,5 +733,7 @@ Standard_Boolean Geom2dGcc_Circ2d2TanOn:
     if (TheSame1(Index) == 0) { return Standard_False; }
     else { return Standard_True; }
   }
+#ifndef _MSC_VER
   return Standard_True;
+#endif 
 }
Index: oce/src/Geom2dGcc/Geom2dGcc_Circ2d2TanRad.cxx
===================================================================
--- oce.orig/src/Geom2dGcc/Geom2dGcc_Circ2d2TanRad.cxx
+++ oce/src/Geom2dGcc/Geom2dGcc_Circ2d2TanRad.cxx
@@ -496,7 +496,9 @@ Standard_Boolean Geom2dGcc_Circ2d2TanRad
     if (TheSame1(Index) == 0) { return Standard_False; }
     else { return Standard_True; }
   }
+#ifndef _MSC_VER
   return Standard_True;
+#endif 
 }
 
 Standard_Boolean Geom2dGcc_Circ2d2TanRad::
@@ -512,5 +514,7 @@ Standard_Boolean Geom2dGcc_Circ2d2TanRad
     if (TheSame1(Index) == 0) { return Standard_False; }
     else { return Standard_True; }
   }
+#ifndef _MSC_VER
   return Standard_True;
+#endif 
 }
Index: oce/src/GeomFill/GeomFill.cxx
===================================================================
--- oce.orig/src/GeomFill/GeomFill.cxx
+++ oce/src/GeomFill/GeomFill.cxx
@@ -579,7 +579,9 @@ Standard_Boolean GeomFill::GetCircle(con
     }
     return Standard_True;
   }
+#ifndef _MSC_VER
   return Standard_False;
+#endif
 }
 
 Standard_Boolean GeomFill::GetCircle(const Convert_ParameterisationType  TConv, 
@@ -763,5 +765,8 @@ Standard_Boolean GeomFill::GetCircle(con
     }
     return Standard_True;
   }
+
+#ifndef _MSC_VER
   return Standard_False;
+#endif
 }
Index: oce/src/GeometryTest/GeometryTest_ConstraintCommands.cxx
===================================================================
--- oce.orig/src/GeometryTest/GeometryTest_ConstraintCommands.cxx
+++ oce/src/GeometryTest/GeometryTest_ConstraintCommands.cxx
@@ -382,7 +382,9 @@ static Standard_Integer cirtang (Draw_In
       return 1;
     }
   }
+#ifndef _MSC_VER // MSVC unreachable warning
   return 0;
+#endif
 }
 
 
Index: oce/src/IntPatch/IntPatch_LineConstructor.cxx
===================================================================
--- oce.orig/src/IntPatch/IntPatch_LineConstructor.cxx
+++ oce/src/IntPatch/IntPatch_LineConstructor.cxx
@@ -262,7 +262,9 @@ static Standard_Real LocalFirstParameter
       return firstp;
     }
   }
+#ifndef _MSC_VER
   return firstp;
+#endif
 }
 
 //=======================================================================
@@ -344,7 +346,9 @@ static Standard_Real LocalLastParameter
       return lastp;
     }
   }
+#ifndef _MSC_VER
   return lastp;
+#endif
 }
 
 
Index: oce/src/IntSurf/IntSurf_Quadric.cxx
===================================================================
--- oce.orig/src/IntSurf/IntSurf_Quadric.cxx
+++ oce/src/IntSurf/IntSurf_Quadric.cxx
@@ -265,7 +265,9 @@ gp_Pnt IntSurf_Quadric::Value(const Stan
     break;
   }
 // pop : pour NT
+#ifndef _MSC_VER
   return gp_Pnt(0,0,0);
+#endif
 }
 // ============================================================
 void IntSurf_Quadric::D1(const Standard_Real U,
@@ -316,7 +318,9 @@ gp_Vec IntSurf_Quadric::DN(const Standar
     break;
   }
 // pop : pour NT
+#ifndef _MSC_VER
   return gp_Vec(0,0,0);
+#endif
 }
 // ============================================================
 gp_Vec IntSurf_Quadric::Normale(const Standard_Real U,
@@ -351,7 +355,9 @@ gp_Vec IntSurf_Quadric::Normale(const St
     break;
   }
 // pop : pour NT
+#ifndef _MSC_VER
   return gp_Vec(0,0,0);
+#endif
 }
 // ============================================================
 gp_Vec IntSurf_Quadric::Normale (const gp_Pnt& P) const
@@ -397,7 +403,9 @@ gp_Vec IntSurf_Quadric::Normale (const g
     }     break;
   }
 // pop : pour NT
+#ifndef _MSC_VER
   return gp_Vec(0,0,0);
+#endif
 }
 // ============================================================
 void IntSurf_Quadric::Parameters (const gp_Pnt& P,
Index: oce/src/MAT2d/MAT2d_Circuit.cxx
===================================================================
--- oce.orig/src/MAT2d/MAT2d_Circuit.cxx
+++ oce/src/MAT2d/MAT2d_Circuit.cxx
@@ -825,7 +825,9 @@ static Standard_Boolean IsSharpCorner (c
   else {
     return Standard_True;
   }
+#ifndef _MSC_VER
   return Standard_False;
+#endif
 }
 
 
Index: oce/src/MNaming/MNaming_NamingRetrievalDriver.cxx
===================================================================
--- oce.orig/src/MNaming/MNaming_NamingRetrievalDriver.cxx
+++ oce/src/MNaming/MNaming_NamingRetrievalDriver.cxx
@@ -50,7 +50,7 @@ static TopAbs_ShapeEnum IntegerToShapeEn
     case  6 : return TopAbs_EDGE;
     case  7 : return TopAbs_VERTEX;
       default :
-	return TopAbs_SHAPE;
+	break;
     }
   return TopAbs_SHAPE;
 }
Index: oce/src/MNaming/MNaming_NamingRetrievalDriver_1.cxx
===================================================================
--- oce.orig/src/MNaming/MNaming_NamingRetrievalDriver_1.cxx
+++ oce/src/MNaming/MNaming_NamingRetrievalDriver_1.cxx
@@ -52,7 +52,7 @@ static TopAbs_ShapeEnum IntegerToShapeEn
     case  6 : return TopAbs_EDGE;
     case  7 : return TopAbs_VERTEX;
       default :
-	return TopAbs_SHAPE;
+	break;
     }
   return TopAbs_SHAPE;
 }
Index: oce/src/ProjLib/ProjLib_ComputeApproxOnPolarSurface.cxx
===================================================================
--- oce.orig/src/ProjLib/ProjLib_ComputeApproxOnPolarSurface.cxx
+++ oce/src/ProjLib/ProjLib_ComputeApproxOnPolarSurface.cxx
@@ -1193,11 +1193,13 @@ Handle(Adaptor2d_HCurve2d)
 //  Modified by Sergey KHROMOV - Thu Apr 18 10:57:51 2002 End
     return Handle(Adaptor2d_HCurve2d)();
   }
+#ifndef _MSC_VER
   myProjIsDone = Standard_False;
 //  Modified by Sergey KHROMOV - Thu Apr 18 10:58:01 2002 Begin
 //   Standard_NoSuchObject_Raise_if(1,"ProjLib_ComputeOnPS: build echec");
 //  Modified by Sergey KHROMOV - Thu Apr 18 10:58:02 2002 End
   return Handle(Adaptor2d_HCurve2d)();
+#endif
 }
 
 
Index: oce/src/ShapeAnalysis/ShapeAnalysis.cxx
===================================================================
--- oce.orig/src/ShapeAnalysis/ShapeAnalysis.cxx
+++ oce/src/ShapeAnalysis/ShapeAnalysis.cxx
@@ -249,7 +249,9 @@ Standard_Boolean ShapeAnalysis::IsOuterB
     Standard_Boolean rescl = (fcl.PerformInfinitePoint () == TopAbs_OUT);
     return rescl;
   }
+#ifndef _MSC_VER
   return Standard_True;
+#endif
 }
 
 //=======================================================================
Index: oce/src/ShapeBuild/ShapeBuild_Edge.cxx
===================================================================
--- oce.orig/src/ShapeBuild/ShapeBuild_Edge.cxx
+++ oce/src/ShapeBuild/ShapeBuild_Edge.cxx
@@ -566,7 +566,9 @@ Handle(Geom2d_Curve) ShapeBuild_Edge::Tr
     }
     return aBSpline2d;
   }
+#ifndef _MSC_VER
   return result;
+#endif
 }
 
 //=======================================================================
Index: oce/src/ShapeConstruct/ShapeConstruct_ProjectCurveOnSurface.cxx
===================================================================
--- oce.orig/src/ShapeConstruct/ShapeConstruct_ProjectCurveOnSurface.cxx
+++ oce/src/ShapeConstruct/ShapeConstruct_ProjectCurveOnSurface.cxx
@@ -1564,7 +1564,9 @@ Handle(Geom_Curve) ShapeConstruct_Projec
 #endif
     return Standard_False;
   }
+#ifndef _MSC_VER
   return Standard_False;  // ramasse-miette
+#endif
 }
 
 /* S4135 : BestExtremum is commented after IsAnIsoparametric works with Precision::Confusion()
Index: oce/src/ShapeCustom/ShapeCustom_BSplineRestriction.cxx
===================================================================
--- oce.orig/src/ShapeCustom/ShapeCustom_BSplineRestriction.cxx
+++ oce/src/ShapeCustom/ShapeCustom_BSplineRestriction.cxx
@@ -760,7 +760,9 @@ Standard_Boolean ShapeCustom_BSplineRest
   //else 
   //Surface is not BSpline or Bezier
   //  return Standard_False;
+#ifndef _MSC_VER
   return Standard_True;
+#endif
 }
 
 //=======================================================================
Index: oce/src/ShapeFix/ShapeFix_EdgeProjAux.cxx
===================================================================
--- oce.orig/src/ShapeFix/ShapeFix_EdgeProjAux.cxx
+++ oce/src/ShapeFix/ShapeFix_EdgeProjAux.cxx
@@ -217,7 +217,9 @@ static Standard_Boolean FindParameterWit
 #endif
     return Standard_False;
   }
+#ifndef _MSC_VER
   return Standard_False;  // normalement, on n y passe jamais
+#endif
 }
 
 //=======================================================================
Index: oce/src/ShapeFix/ShapeFix_Wireframe.cxx
===================================================================
--- oce.orig/src/ShapeFix/ShapeFix_Wireframe.cxx
+++ oce/src/ShapeFix/ShapeFix_Wireframe.cxx
@@ -386,7 +386,6 @@ ShapeFix_Wireframe::ShapeFix_Wireframe(c
     B.SameRange(newedge,Standard_False);
     
     E3 = newedge;
-    return ReplaceFirst;
   }
   catch ( Standard_Failure ) 
   {
@@ -394,7 +393,6 @@ ShapeFix_Wireframe::ShapeFix_Wireframe(c
     cout<<"Error: ShapeFix_Wireframe::FixSmallEdges: JoinEdges: Exception in GeomConvert_CompCurveToBSplineCurve: ";
     Standard_Failure::Caught()->Print(cout); cout<<endl;
 #endif
-    return ReplaceFirst;
   }
   return ReplaceFirst;
 }
Index: oce/src/StdPrs/StdPrs_DeflectionCurve.cxx
===================================================================
--- oce.orig/src/StdPrs/StdPrs_DeflectionCurve.cxx
+++ oce/src/StdPrs/StdPrs_DeflectionCurve.cxx
@@ -278,7 +278,9 @@ static Standard_Boolean MatchCurve (
       }
     }
   }
+#ifndef _MSC_VER
   return Standard_False;
+#endif
 }
 
 
Index: oce/src/TopOpeBRep/TopOpeBRep_FFDumper.cxx
===================================================================
--- oce.orig/src/TopOpeBRep/TopOpeBRep_FFDumper.cxx
+++ oce/src/TopOpeBRep/TopOpeBRep_FFDumper.cxx
@@ -258,6 +258,8 @@ void TopOpeBRep_FFDumper::DumpVP(const T
 #ifndef DEB
 Standard_Integer TopOpeBRep_FFDumper::ExploreIndex(const TopoDS_Shape& , const Standard_Integer ) const
 {
+	return 0;
+}
 #else
 Standard_Integer TopOpeBRep_FFDumper::ExploreIndex(const TopoDS_Shape& S, const Standard_Integer ISI) const
 {
@@ -265,9 +267,8 @@ Standard_Integer TopOpeBRep_FFDumper::Ex
   if (ISI == 1) r = myEM1.Find(S);
   if (ISI == 2) r = myEM2.Find(S);
   return r;
-#endif
-  return 0;
 }
+#endif
 
 //=======================================================================
 //function : DumpDSP
Index: oce/src/TopOpeBRep/TopOpeBRep_sort.cxx
===================================================================
--- oce.orig/src/TopOpeBRep/TopOpeBRep_sort.cxx
+++ oce/src/TopOpeBRep/TopOpeBRep_sort.cxx
@@ -38,7 +38,7 @@ int compll(const void* v1, const void* v
   Standard_Real p2 = i2->Parameter();
   if      (p1 < p2) return (-1);
   else if (p1 > p2) return (1);
-  else return (0);
+  
   return (0);
 }
      
Index: oce/src/TopOpeBRep/TopOpeBRep_traceBOOPNINT.cxx
===================================================================
--- oce.orig/src/TopOpeBRep/TopOpeBRep_traceBOOPNINT.cxx
+++ oce/src/TopOpeBRep/TopOpeBRep_traceBOOPNINT.cxx
@@ -59,10 +59,8 @@ Standard_Integer BOOPNINTL::Set(const St
     mynl++;
     return 0;
   }
-  else {
-    return 1;
-  }
-  return 0;
+ 
+   return 1;
 }
 
 Standard_Boolean BOOPNINTL::Get(Standard_Integer n,char**a){
Index: oce/src/TopOpeBRepBuild/TopOpeBRepBuild_Builder1.cxx
===================================================================
--- oce.orig/src/TopOpeBRepBuild/TopOpeBRepBuild_Builder1.cxx
+++ oce/src/TopOpeBRepBuild/TopOpeBRepBuild_Builder1.cxx
@@ -1778,7 +1778,10 @@ Standard_Integer TopOpeBRepBuild_Builder
     // Unknowm case for existing adjacents
     return 0;
   }
+ 
+#ifndef _MSC_VER
   return -1;
+#endif
 }
 
 //=======================================================================
Index: oce/src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx
===================================================================
--- oce.orig/src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx
+++ oce/src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx
@@ -367,8 +367,9 @@ Standard_Boolean TopOpeBRepBuild_GTopo::
     else IsToRev = (s1 == TopAbs_IN);
     return IsToRev;
   }
-  Standard_ProgramError::Raise("GTopo::IsToReverse1");
+#ifndef _MSC_VER
   return Standard_False; // dummy
+#endif
 }
 
 
@@ -390,8 +391,9 @@ Standard_Boolean TopOpeBRepBuild_GTopo::
     else IsToRev = (s2 == TopAbs_IN);
     return IsToRev;
   }
-  Standard_ProgramError::Raise("GTopo::IsToReverse2");
+#ifndef _MSC_VER
   return Standard_False; // dummy
+#endif
 }
 
 //=======================================================================
Index: oce/src/TopOpeBRepBuild/TopOpeBRepBuild_Tools_1.cxx
===================================================================
--- oce.orig/src/TopOpeBRepBuild/TopOpeBRepBuild_Tools_1.cxx
+++ oce/src/TopOpeBRepBuild/TopOpeBRepBuild_Tools_1.cxx
@@ -257,10 +257,7 @@ void CorrectEdgeTolerance (const TopoDS_
 	GC->Range(f,l);
 	if (SameRange && (f != First || l != Last)) {
 	  return ;//BRepCheck_InvalidSameRangeFlag);
-	  if (SameParameter) {
-	    return; //BRepCheck_InvalidSameParameterFlag);
-	  }
-        }
+       }
 	
 	Handle(Geom_Surface) Sb = cr->Surface();
 	Sb = Handle(Geom_Surface)::DownCast (Su->Transformed(L.Transformation()));
Index: oce/src/TopOpeBRepDS/TopOpeBRepDS.cxx
===================================================================
--- oce.orig/src/TopOpeBRepDS/TopOpeBRepDS.cxx
+++ oce/src/TopOpeBRepDS/TopOpeBRepDS.cxx
@@ -135,7 +135,7 @@ Standard_Boolean TopOpeBRepDS::IsTopolog
   case TopOpeBRepDS_WIRE    : return Standard_True; break;
   case TopOpeBRepDS_EDGE    : return Standard_True; break;
   case TopOpeBRepDS_VERTEX  : return Standard_True; break;
-  default : return Standard_False;
+  default : break;
   }
   return Standard_False;
 }
Index: oce/src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx
===================================================================
--- oce.orig/src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx
+++ oce/src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx
@@ -165,9 +165,7 @@ static Standard_Boolean GetOrigin(const
     o = Handle(Geom2d_Line)::DownCast(PC)->Location();
     return Standard_True;
   }
-  else  {
-    return Standard_False;
-  }
+ 
   return Standard_False;
 }
 #endif
@@ -196,9 +194,7 @@ static Standard_Boolean GetOrigin(const
     o = Handle(Geom_Line)::DownCast(C)->Position().Location();
     return Standard_True;
   }
-  else  {
-    return Standard_False;
-  }
+ 
   return Standard_False;
 }
 #endif
Index: oce/src/TopOpeBRepDS/TopOpeBRepDS_PointIterator.cxx
===================================================================
--- oce.orig/src/TopOpeBRepDS/TopOpeBRepDS_PointIterator.cxx
+++ oce/src/TopOpeBRepDS/TopOpeBRepDS_PointIterator.cxx
@@ -100,7 +100,9 @@ Standard_Real  TopOpeBRepDS_PointIterato
     Standard_ProgramError::Raise("TopOpeBRepDS_PointIterator::Parameter()");
     return 0.;  // windowsNT
   }
+#ifndef _MSC_VER
   return 0.;  // windowsNT
+#endif
 }
 
 //=======================================================================
@@ -139,7 +141,9 @@ Standard_Boolean TopOpeBRepDS_PointItera
     Standard_ProgramError::Raise("TopOpeBRepDS_PointIterator::DiffOriented()");
     return Standard_False; // windowsNT
   }
+#ifndef _MSC_VER
   return Standard_False; // windowsNT
+#endif
 }
 
 //=======================================================================
@@ -158,7 +162,9 @@ Standard_Boolean TopOpeBRepDS_PointItera
     Standard_ProgramError::Raise("TopOpeBRepDS_PointIterator::SameOriented()");
     return Standard_False; // windowsNT
   }
+#ifndef _MSC_VER
   return Standard_False; // windowsNT
+#endif
 }
 
 //=======================================================================
Index: oce/src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx
===================================================================
--- oce.orig/src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx
+++ oce/src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx
@@ -182,7 +182,6 @@ static Standard_Integer FUN_mkT2dquad(co
   if (samec)           return isON2ifss;//is ON if curves are on same side/tg line
   if (curvC1 > curvC2) return isIN2ifss;//is IN if curves are on same side/tg line
   else                 return isOU2ifss;//is OU if curves are on same side/tg line
-  return 0;
 }
 
 static Standard_Boolean FUN_getnearpar(const TopoDS_Edge& e,const Standard_Real par,const Standard_Real f,const Standard_Real l,
@@ -255,7 +254,9 @@ static Standard_Boolean FUN_getsta(const
     sta = TopAbs_ON; 
     return Standard_True;
   } 
+#ifndef _MSC_VER
   return Standard_False;
+#endif
 } 
 static Standard_Boolean FUN_mkT2dquad(const TopoDS_Edge& e1,const Standard_Real par1,const Standard_Real f1,const Standard_Real l1,
 			 const TopoDS_Edge& e2,const Standard_Real par2,
Index: oce/src/XSControl/XSControl_Functions.cxx
===================================================================
--- oce.orig/src/XSControl/XSControl_Functions.cxx
+++ oce/src/XSControl/XSControl_Functions.cxx
@@ -140,7 +140,10 @@ static IFSelect_ReturnStatus XSControl_x
     IFSelect_Activator::SetCurrentAlias (WS->SelectedNorm(Standard_True));
     return IFSelect_RetDone;
   }
+
+#ifndef _MSC_VER // Unreachable code warning in MSVC
   return IFSelect_RetVoid;
+#endif
 }
 
 
